src/server.rs

```diff
@@ -12,21 +12,28 @@ use crate::bytes::{FromBytes, IntoBytes};
 use crate::connection::Connection;
 use crate::connection::MIN_PORT_NUMBER;
 use crate::packet::*;
+use crate::RetransmissionConfig;
 
 /// A TFTP server.
 pub struct Server {
     socket: UdpSocket,
     serve_dir: PathBuf,
+    retransmission_config: Option<RetransmissionConfig>,
 }
 
 impl Server {
     /// Creates a server configured to serve files from a given directory on
     /// a given address.
-    pub fn new<A: ToSocketAddrs, P: AsRef<Path>>(bind_to: A, serve_from: P) -> Result<Self> {
+    pub fn new<A: ToSocketAddrs, P: AsRef<Path>>(
+        bind_to: A,
+        serve_from: P,
+        retransmission_config: Option<RetransmissionConfig>,
+    ) -> Result<Self> {
```

Comment:
```suggestion
    pub fn new<A: ToSocketAddrs, P: AsRef<Path>>(
        bind_to: A,
        serve_from: P,
    ) -> Result<Self> {
```

Similar feedback to `Client`. Once `RetransmissionConfig` implements `Default`, let's just have this constructor use its default value.